include/mata/nfa/plumbing.hh

```diff
@@ -54,21 +56,22 @@ inline void determinize(Nfa* result, const Nfa& aut, std::unordered_map<StateSet
     *result = determinize(aut, subset_map);
 }
 
-inline void reduce(Nfa* result, const Nfa &aut, bool trim_result = true, StateToStateMap *state_map = nullptr,
-    const StringMap&  params = {{"algorithm", "simulation"}}) { *result = reduce(aut, trim_result, state_map, params); }
+inline void reduce(Nfa* result, const Nfa &aut, bool trim_result = true, StateRenaming *state_renaming = nullptr,
```

Comment:
In the previous comment, I meant these optional parameters.

Reply:
Trim will be removed and replaced with chaining. `state_renaming` is an output parameter which will be handled as output parameter, not as optional input parameter.

Reply:
Trim refactored.